src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -322,7 +359,13 @@ object FileProcessor {
       id: UUID,
       parentId: Option[UUID],
       title: String,
-      name: String
+      name: String,
+      id_UpstreamSystemReference: String,
+      originalFiles: List[UUID],
+      originalMetadataFiles: List[UUID],
+      description: Option[String],
+      id_URI: Option[String],
+      id_NeutralCitation: Option[String]
```

Comment:
Can we have the id_ fields as a `List[IdField]` like it is in the `BagitFolderMetadataObject` Then you can do 
```scala
          idFields.map { idField =>
            (s"id_${idField.name}", Json.fromString(idField.value))
          }
```
The advantage is that we won't need to change the case classes if we ever need more ID fields (which we probably will) and we don't have to have attribute names in the case classes called `id_`
It also shortens the custom decoder a bit.